Code/src/main/java/nl/utwente/group10/ui/components/Connection.java

```diff
@@ -206,4 +215,23 @@ public String toString() {
         return "Connection connecting \n(out) " + startAnchor + "   to\n(in)  "
                 + endAnchor;
     }
+    
+    /**
+     * This method evaluates the validity of the created connection.
+     * If the connection results in an invalid operation a visual
+     * error will be displayed.
+     */
+    public void checkError() {
+        if(startAnchor.isPresent() && endAnchor.isPresent()) {
+            try {
+                //TODO Obviously this will cause errors, we need a way to access the Env
+                endAnchor.get().getBlock().asExpr().analyze(new Env(), new GenSet());
+                this.setStyle("-fx-stroke: black;");
+            } catch (HaskellTypeError e) {
+                this.setStyle("-fx-stroke: red;");
+            } catch (HaskellException e) {
+                //e.printStackTrace();
```

Comment:
Dat is nog steeds een ander probleem dan specifiek deze comment, maar die Env kun je trekken uit HaskellCatalog en die GenSet kun je voor nu gewoon blijven bouwen.